src/OpenPostbud/database/data_types/obfuscated_id.py

```diff
@@ -0,0 +1,67 @@
+"""This module contains a type decorator class for use in ORM models."""
+
+import hashlib
+
+from sqlalchemy import Dialect, types
+
+
+# pylint: disable=too-many-ancestors, abstract-method
+class ObfuscatedId(types.TypeDecorator):
+    """A type decorator used when defining sqlalchemy columns.
+    This decorater automatically obfuscates incremental integer ids.
+    """
+    impl = types.INTEGER
+    cache_ok = True
+
+    def __init__(self, prefix: str):
+        """Initialise a new Obfuscator.
+
+        Args:
+            prefix: The prefix and salt to use on obfuscated ids.
+        """
+        super().__init__()
+        self.prefix = prefix
+
+    # pylint: disable=unused-argument
+    def process_bind_param(self, value: str, dialect: Dialect) -> int:
+        """Convert to incremental int before sending to database."""
+        s = value.replace(self.prefix, "", 1)
+        n = int(s)
+        cipher = _feistel_cipher(n, self.prefix)
+        return cipher
+
+    # pylint: disable=unused-argument
+    def process_result_value(self, value: int, dialect: Dialect) -> str:
+        """Convert to obfuscated string when fetching from database."""
+        cipher = _feistel_cipher(value, self.prefix)
+        return self.prefix + str(cipher)
+
+
+def _feistel_cipher(number: int, salt: str) -> int:
```

Comment:
Way too much complexity for making "nice ID's" in a front end. It will make debugging harder and can be a course for errors